rt/rs/extensions/reactor/src/main/java/org/apache/cxf/jaxrs/reactor/client/FluxRxInvokerImpl.java

```diff
@@ -0,0 +1,178 @@
+    }
+
+    @Override
+    public <R> Flux<R> put(Entity<?> entity, GenericType<R> genericType) {
+        return method(HttpMethod.PUT, entity, genericType);
+    }
+
+    @Override
+    public Flux<Response> post(Entity<?> entity) {
+        return method(HttpMethod.POST, entity);
+    }
+
+    @Override
+    public <R> Flux<R> post(Entity<?> entity, Class<R> responseType) {
+        return method(HttpMethod.POST, entity, responseType);
+    }
+
+    @Override
+    public <R> Flux<R> post(Entity<?> entity, GenericType<R> genericType) {
+        return method(HttpMethod.POST, entity, genericType);
+    }
+
+    @Override
+    public Flux<Response> delete() {
+        return method(HttpMethod.DELETE);
+    }
+
+    @Override
+    public <R> Flux<R> delete(Class<R> responseType) {
+        return method(HttpMethod.DELETE, responseType);
+    }
+
+    @Override
+    public <R> Flux<R> delete(GenericType<R> genericType) {
+        return method(HttpMethod.DELETE, genericType);
+    }
+
+    @Override
+    public Flux<Response> head() {
+        return method(HttpMethod.HEAD);
+    }
+
+    @Override
+    public Flux<Response> options() {
+        return method(HttpMethod.OPTIONS);
+    }
+
+    @Override
+    public <R> Flux<R> options(Class<R> responseType) {
+        return method(HttpMethod.OPTIONS, responseType);
+    }
+
+    @Override
+    public <R> Flux<R> options(GenericType<R> genericType) {
+        return method(HttpMethod.OPTIONS, genericType);
+    }
+
+    @Override
+    public Flux<Response> trace() {
+        return method(TRACE);
+    }
+
+    @Override
+    public <R> Flux<R> trace(Class<R> responseType) {
+        return method(TRACE, responseType);
+    }
+
+    @Override
+    public <R> Flux<R> trace(GenericType<R> genericType) {
+        return method(TRACE, genericType);
+    }
+
+    @Override
+    public Flux<Response> method(String name) {
+        return method(name, Response.class);
+    }
+
+    @Override
+    public <R> Flux<R> method(String name, Class<R> responseType) {
+        return flux(webClient.async().method(name, responseType));
+    }
+
+    @Override
+    public <R> Flux<R> method(String name, GenericType<R> genericType) {
+        return flux(webClient.async().method(name, genericType));
+    }
+
+    @Override
+    public Flux<Response> method(String name, Entity<?> entity) {
+        return method(name, entity, Response.class);
+    }
+
+    @Override
+    public <R> Flux<R> method(String name, Entity<?> entity, Class<R> responseType) {
+        return flux(webClient.async().method(name, entity, responseType));
+    }
+
+    @Override
+    public <R> Flux<R> method(String name, Entity<?> entity, GenericType<R> genericType) {
+        return flux(webClient.async().method(name, entity, genericType));
+    }
+
+    private <R> Flux<R> flux(Future<R> future) {
+        Flux<R> flux = Flux.from(Mono.fromFuture(toCompletableFuture(future, executorService)));
+        if (scheduler != null) {
```

Comment:
@johnament Thanks for passing the `executorService` to `toCompletableFuture`. The `scheduler` and  this snippet:
```
if (scheduler != null) {
            flux = flux.subscribeOn(scheduler);
}
```
are not needed. The reason is that the completion callback for `Mono.fromFuture` will be called in one of the `executorService` threads or `ForkJoin` pool (you could easily trace it). Moreover, we could run into issues with this code because it uses two threads from the same pool for `toCompletableFuture` and subscription, exposing ourselves for likelihood of livelocks.

Reply:
Since the underlying call is happening via async webclient, we can't take full advantage of the reactive nature (but not sure the JSON format works well, i want to play around with SSE next after this is merged).  Scheduler should not be used.